### DISCUSSION TITLE
Episode 167: Phone Tag

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#59] [Episode]
Episode Link: https://adspthepodcast.com/2024/02/02/Episode-167.html

Feel free to write comments or ask questions 🙂

**Comment by robertFrysch** [comment]
Hi Conor, hi Bryce,

it was very cool that you chose the phone tag format and didn't leave out an episode. I also found the result was surprisingly well done, thank you! However, it was a bit denser and less chaotic than usual this time 😉

Among other things, Bryce addressed a topic that I found particularly interesting: namely the integration of array languages into "industrially used languages" such as C++. The pipe syntax of ranges is a successful example of this. However, I'm also wondering to what extent the stack-based composition of algorithms (from functions and combinators) that [Uiua](https://www.uiua.org/) provides can be integrated into C++, because I find it a very powerful alternative of creating complex algorithms (e.g. it's possible to handle more than on data/range at a time). Over the holidays, I played around with this and wrote a small example implementation of what I mean:
https://gitlab.com/robertFrysch/alco/
I know that the interface is not quite optimal yet, but I wanted to ask you and the community what you think of such an approach.

In the examples, I have recreated some of the solutions from Conor's videos:
https://gitlab.com/robertFrysch/alco/-/blob/main/examples.cpp

I also created an example for the "Trapping Rain Water", which you can find on Compiler Explorer:
https://godbolt.org/z/Gxxrh3Y4o

Best regards and keep up the good job,
Robert

**Reply by JVApen** [reply]
Really nice to have a non-chaotic episode. I was really surprised to hear how long Bryce was talking without deviation of topic.
